src/Italy/Mews.Fiscalizations.Italy.Tests/UniwixClientTests.cs

```diff
@@ -0,0 +1,218 @@
+﻿using NUnit.Framework;
+using System;
+using System.Threading.Tasks;
+using System.Linq;
+using Mews.Fiscalizations.Italy.Dto.Invoice;
+using Mews.Fiscalizations.Core.Model;
+using Mews.Fiscalizations.Italy.Constants;
+using System.Threading;
+using Mews.Fiscalizations.Italy.Uniwix.Communication;
+using Mews.Fiscalizations.Italy.Uniwix.Communication.Dto;
+
+namespace Mews.Fiscalizations.Italy.Tests
+{
+    [TestFixture]
+    public sealed class UniwixClientTests
+    {
+        public static readonly string Username = Environment.GetEnvironmentVariable("italian_username") ?? "INSERT_USERNAME";
+        public static readonly string Password = Environment.GetEnvironmentVariable("italian_password") ?? "INSERT_PASSWORD";
+
+        public static UniwixClient GetUniwixClient()
+        {
+            return new UniwixClient(new UniwixClientConfiguration(Username, Password));
+        }
+
+        [Test]
+        public async Task SendInvoiceSucceeds()
+        {
+            var client = GetUniwixClient();
+            var invoiceNumber = new Random().Next(1, 9999).ToString();
+            var result = await client.SendInvoiceAsync(new ElectronicInvoice
+            {
+                Version = VersioneSchemaType.FPR12,
+                Header = GetInvoiceHeader(invoiceNumber),
+                Body = new[] { GetInvoiceBody(invoiceNumber) }
```

Comment:
Won't this complain about duplicate invoices? Can't we make it like in other libraries, where the date is present in the invoice number?

Reply:
Seems like it won't in the test environment as the state will be always "Uploading file..." (which is pending).